src/lang/proc_macros/plugins/downcast.rs

```diff
@@ -14,10 +14,12 @@ use crate::lang::db::AnalysisDatabase;
 /// Safety: This function MUST only be invoked with an object that is of type
 /// [AnalysisDatabase]. Using it with any other type leads to undefined behavior.
 pub(super) unsafe fn unsafe_downcast_ref(db: &dyn SyntaxGroup) -> &AnalysisDatabase {
-    // Replicated logic from `impl dyn Any downcast_ref_unchecked()`.
-    // This approach works as long as `impl dyn Any downcast_ref_unchecked()` implementation is
-    // unchanged and the caller can ensure that `db` is truly an instance of AnalysisDatabase.
-    &*(db as *const dyn SyntaxGroup as *const AnalysisDatabase)
+    unsafe {
```

Comment:
did it become unsafe in 2024's edition?

Reply:
this is edition change: https://blog.rust-lang.org/2025/02/20/Rust-1.85.0.html